django_project/ta_system/admin.py

```diff
@@ -131,6 +140,20 @@ def assignment_data_download(self, request):
                 )
         return redirect('admin:index')
 
+    def change_system_status(self, request):
+        if request.method != 'POST':
+            return handle_bad_request(request, app='admin', expected_method='POST')
+        if models.SystemStatus.objects.count() > 0:
+            previous_system_status = models.SystemStatus.objects.order_by(
+                'id').last()
```

Comment:
let's move this back to one line, but more importantly, is it possible to order by the 'date_changed' field instead? Ordering by the 'id' may work for now since (i think) Django uses an auto-increment field. But semantically, we really want to order by the 'date_changed', since we're trying to grab the most recent system status.

Reply:
saved line 147-148 into one line. I think order by id is better as the next id will always be the previous id+1